### PR TITLE
Refactor Errors

### PR DESCRIPTION
**Pull request by marceloneil**
Default to `ChatDebug` instead of just `Debug`

### CODE REVIEW

**Comment by marceloneil** [comment]
Going through the errors, I didn't change anything but there are many instances were the issue isn't but should be surfaced to the user. For example, the webhookbot will say "Something went wrong!" if the user is not an admin.

File: githubbot/githubbot/handler.go [hunk 1]
```diff
@@ -132,112 +127,101 @@ func (h *Handler) handleSubscribe(cmd string, msg chat1.MsgSummary, create bool,
...
 	toks, err := shellquote.Split(cmd)
 	if err != nil {
-		h.Debug("error splitting command: %s", err)
-		return
+		return fmt.Errorf("error splitting command: %s", err.Error())
 	}
 	args := toks[2:]
 	var message string
-	defer func() {
+	defer func() error {
```
[the remark this answers is not in the document]
Comment (marceloneil) [reply githubbot/githubbot/handler.go]: so something like this:
```
func handleWatch() (err error) {
  ...
  defer func() error {
    err = fmt.Errorf()
  }
  ...
}
```
should work?

[the remark this answers is not in the document]
Comment (joshblum) [reply githubbot/githubbot/handler.go]: yup!

[the remark this answers is not in the document]
Comment (marceloneil) [reply githubbot/githubbot/handler.go]: alright let me know if my fix works

[the remark this answers is not in the document]
Comment (marceloneil) [reply githubbot/githubbot/handler.go]: I think it should, I tested out a tiny defer function

**Comment by joshblum** [comment]
@marceloneil if you rebase master CI should pass

File: githubbot/githubbot/handler.go [hunk 2]
```diff
@@ -67,57 +67,49 @@ func (h *Handler) HandleCommand(msg chat1.MsgSummary) error {
...
 
 	var message string
 	defer func() {
 		if message != "" {
 			_, err = h.kbc.SendMessageByConvID(msg.ConvID, fmt.Sprintf(message, args[0]))
-			if err != nil {
-				h.ChatDebug(msg.ConvID, "Error sending message: %s", err)
-				return
-			}
+			err = fmt.Errorf("error sending message: %s", err)
```
Comment (joshblum) [inline githubbot/githubbot/handler.go]: You only want to assign this if `err != nil`, otherwise we will always respond with this error message

Comment (marceloneil) [reply githubbot/githubbot/handler.go]: Oh good catch 🤦‍♂

**Pull request merged**